src/clinical/service-worker-thread/tasks.ts

```diff
@@ -107,11 +107,11 @@ const sortDocs = (sort: string, entityName: string, completionStats: CompletionR
 
   // -1 Current lower index than Next, +1 Current higher index than Next
   let order = 0;
-  const isDescending = sort.includes('-') ? -1 : 1;
+  const isDescending = sort.startsWith('-') ? -1 : 1;
 
   if (isDefaultSort) {
-    const { donorId: currentDonorId } = currentRecord as Donor;
-    const { donorId: nextDonor } = nextRecord as Donor;
+    const { donorId: currentDonorId } = currentRecord;
+    const { donorId: nextDonor } = nextRecord;
```

Comment:
we want to do the same on line 128 too I think?
apparently i can't comment on unchanged lines in a PR :(

Reply:
Thanks for the approval @ciaranschutte 
I ended up refactoring this significantly as part of the investigation for #[2546](https://app.zenhub.com/workspaces/icgc-argo-platform-dk-production-board-5e542d38415f5034e9fed89d/issues/gh/icgc-argo/platform-ui/2546) to make the logic more straightforward
Sorting has been broken up into multiple functions
And using `Donor | ClinicalInfo` was causing cascading Typescript issues which were a major pain, so for now I've simplified it to only `ClinicalInfo`